extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/GeneratedStaticResourceBuildItem.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.vertx.http.deployment.spi;
+
+import io.quarkus.builder.item.MultiBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+
+/**
+ * This build item aims to be used by extensions to generate static resources.
+ * <br>
+ * Those resources will be served on the given {@link GeneratedStaticResourceBuildItem#path}. It is NOT necessary to create the
+ * file on disk.
+ * <br>
+ * Behind the scenes the build step will take care of add those resources to the final build, through
+ * {@link AdditionalStaticResourceBuildItem}, {@link NativeImageResourceBuildItem}
+ * and {@link io.quarkus.deployment.builditem.GeneratedResourceBuildItem} build items.
+ * <br>
+ * The value of {@code path} should <b>NOT</b> be prefixed with {@code '/'} and is assumed to be a path
+ * under {@code 'META-INF/resources'} on classpath.
+ */
+public final class GeneratedStaticResourceBuildItem extends MultiBuildItem {
+
+    private final String path;
+    private final byte[] content;
+
+    public GeneratedStaticResourceBuildItem(final String path, final byte[] content) {
+        this.path = path;
+        this.content = content;
+    }
+
+    public String getPath() {
+        return this.path;
+    }
+
+    public byte[] getContent() {
```

Comment:
It's already the case with GeneratedResourceBuildItem right?

Reply:
Yes, but we need to make sure we do not run out of memory

Reply:
@cescoffier how could we do that? We could add a size limit? But what would be a good max value?

Reply:
Generated Resource are all resources which where not usable as is from the sources, so it could include image and other things.

After thinking a bit more about it, this is just an way in the GeneratedResourceBuildItem which already exist, we should add a limit in there if we really want to, but I think we are fine since in prod there is a cache as we are not changing the flow at all.

In dev, the problem is just for loading those resources, in a follow up PR I will add a cache, which gets cleared after a successfull doScan. 

This way we have a proper solution for all cases.

Reply:
One thing we might generate is a JS API for REST endpoints. It could be large. As long as they end up on disk, I don't think we need to have a size limit.